src/transformers/models/jina_bert/modular_jina_bert.py

```diff
@@ -0,0 +1,1169 @@
+            logger.warning(
+                "If you want to use `JinaBertForMaskedLM` make sure `config.is_decoder=False` for "
+                "bi-directional self-attention."
+            )
+
+        self.bert = JinaBertModel(config, add_pooling_layer=False)
+        self.cls = JinaBertOnlyMLMHead(config)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+
+class JinaBertForNextSentencePrediction(BertForNextSentencePrediction):
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.bert = JinaBertModel(config)
+        self.cls = JinaBertOnlyNSPHead(config)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+
+class JinaBertForSequenceClassification(BertForSequenceClassification):
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+        self.config = config
+
+        self.bert = JinaBertModel(config)
+        classifier_dropout = (
+            config.classifier_dropout if config.classifier_dropout is not None else config.hidden_dropout_prob
+        )
+        self.dropout = nn.Dropout(classifier_dropout)
+        self.classifier = nn.Linear(config.hidden_size, config.num_labels)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+
+class JinaBertForMultipleChoice(BertForMultipleChoice):
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.bert = JinaBertModel(config)
+        classifier_dropout = (
+            config.classifier_dropout if config.classifier_dropout is not None else config.hidden_dropout_prob
+        )
+        self.dropout = nn.Dropout(classifier_dropout)
+        self.classifier = nn.Linear(config.hidden_size, 1)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+
+class JinaBertForTokenClassification(BertForTokenClassification):
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+
+        self.bert = JinaBertModel(config, add_pooling_layer=False)
+        classifier_dropout = (
+            config.classifier_dropout if config.classifier_dropout is not None else config.hidden_dropout_prob
+        )
+        self.dropout = nn.Dropout(classifier_dropout)
+        self.classifier = nn.Linear(config.hidden_size, config.num_labels)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+
+class JinaBertForQuestionAnswering(BertForQuestionAnswering):
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+
+        self.bert = JinaBertModel(config, add_pooling_layer=False)
+        self.qa_outputs = nn.Linear(config.hidden_size, config.num_labels)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+
+# TODO why do I need this but other modular implementations don't have it
```

Comment:
Alright, checked again and I can find the same line in [this PR](https://github.com/huggingface/transformers/pull/35224/files#diff-2b3fa28264c0c6f7294d5208c2ac55d5b48c4a74d9c7f9ba411722a2eab6f634R744). So I guess it's fine